std/utf.d

```diff
@@ -2420,6 +2469,167 @@ unittest // bugzilla 12923
     }());
 }
 
+/++
+    Returns whether the given range of characters is well-formed Unicode or not.
+
+    Params:
+        range = The range of charaters to validate.
+        index = The index (in code units) where the failed character was (or `0`
+                if the result is `true`).
+  +/
+pragma(inline, true) bool isValidUnicode(R)(R range)
+    if (isInputRange!R && isSomeChar!(ElementType!R) && !isInfinite!R)
+{
+    size_t failedIndex;
+    return _isValidUnicodeImplTrustWrap!false(range, failedIndex);
+}
+
+/// Ditto
+pragma(inline, true) bool isValidUnicode(R)(R range, out size_t failedIndex)
+    if (isInputRange!R && isSomeChar!(ElementType!R) && !isInfinite!R)
+{
+    return _isValidUnicodeImplTrustWrap!true(range, failedIndex);
+}
+
+pragma(inline, true)
+private bool _isValidUnicodeImplTrustWrap(bool getIndex, R)(R range, out size_t failedIndex) @trusted
```

Comment:
I have submitted an alternative to my earlier PR which also solves the `@trusted` problem: https://github.com/D-Programming-Language/phobos/pull/4073
The diff is kind of large for what it accomplishes, though.